onnxruntime/core/session/inference_session.cc

```diff
@@ -580,6 +596,32 @@ common::Status InferenceSession::ValidateInputs(const std::vector<std::string>&
       auto input_type = input_ml_value.Type();
       ORT_RETURN_IF_ERROR(CheckTypes(input_type, expected_type));
     }
+
+    // IR < 4 all entries in input_def_map_ are required.
+    // IR 4 or above we need to do a lookup to see if this was a required input
+    if (model_ir_version < 4 || required_inputs_.find(feed_name) != required_inputs_.cend()) {
```

Comment:
Would be better if we have a util function, instead of hard- code version 4 in comments...